### PR TITLE
`Plagiarism checks`: Fix plagiarism case verdict message

### PR DESCRIPTION
**Pull request by jakubriegel**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context
Plagiarism case verdict email didn't have its title placeholder resolved.

### Description
This PR adds missing `if` statement for resolving placeholder title of email sent when plagiarism case is resolved.

Surprisingly, there was a test case for that, which asserted that wrong value was returned by Artemis.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 2 Students (on of them needs to be your TUM account, so that email is delivered)
- **test server with email support (example _test1_)**

1. Create an exercise (text is easiest to test). 
2. Submit two identical solution from two different student accounts.
3. From the instructor account go to the exercise management page -> check plagiarism.
4. Run checks and confirm displayed comparison.
5. Go to the new plagiarism case: Course -> Assessment -> Plagiarism cases.
6. Send a default notification to the student (yourself)
7. Select and apply plagiarism case verdict.
8. Verify that you got an email with proper title (not placeholder, see screenshots below)

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
Before:
image

After:
image

### CODE REVIEW

**Comment by jakubriegel** [comment]
This PR is applied on the top of #7302 and should be merged to `develop` after it.

**Comment by xHadie** [comment]
_Tested on Legacy TS3_

I followed your instructions, but when I click on "Check Plagiarism" I get the error that no valid submissions exist. (Text Exercise)
Do I need to test on a specifc server or did I miss anything to make the submissions valid?
Submissions of both students where graded by a course instructor. I also removed or played around with the settings in the plagiarism and exercise view with the hope to make the submissions valid, with no success.

Not sure if this is a bug, maybe you could update the instructions a bit? I'd be happy to test again then.

Screenshot 2023-10-24 121725

**Review by vinceclifford — DISMISSED** [review]
Works as intended. Looks good.

**Review by florian-glombik — APPROVED** [review]
The Mail is displayed properly in Outlook

image

**Comment by jakubriegel** [comment]
> _Tested on Legacy TS3_
> 
> I followed your instructions, but when I click on "Check Plagiarism" I get the error that no valid submissions exist. (Text Exercise) Do I need to test on a specifc server or did I miss anything to make the submissions valid? Submissions of both students where graded by a course instructor. I also removed or played around with the settings in the plagiarism and exercise view with the hope to make the submissions valid, with no success.
> 
> Not sure if this is a bug, maybe you could update the instructions a bit? I'd be happy to test again then.
> 
> Screenshot 2023-10-24 121725

Most likely your submissions were not long enough. By default you need to have 50 words for text exercise or 50 diff lines for programming exercise. You change for each run

**Review by basak-akan — APPROVED** [review]
Code looks good 👍

**Review by vinceclifford — APPROVED** [review]
Reviewed

**Review by milljoniaer — APPROVED** [review]
Tested on ts1, works as expected

**Comment by nata-81** [comment]
> Tested on ts1, works as expected

I am trying to test the case on ts1 and after step 4. Run checks and confirm displayed comparison:

image



I cant find the Plagiarismcases as expected:

image


What am I doing wrong?

Branch: plagiarism-checks-fix-verdict-message •
Commit Hash: 54b4e39 •
Committer Date: Sat, 28 Oct 2023 08:11:39 GMT •
Last Committer: jakubriegel

**Comment by jakubriegel** [comment]
@nata-81 You can find plagiarism cases under the Assessment menu

I've updated PR description to be more informative

**Comment by nata-81** [comment]
Tested on ts1, works as expected.

File: src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java [hunk 1]
```diff
@@ -379,21 +380,23 @@ void testNotifyUserAboutNewPossiblePlagiarismCase() throws MessagingException {
...
         // explicitly change the user to prevent issues in the following server call due to userRepository.getUser() (@WithMockUser is not working here)
         userUtilService.changeUser(TEST_PREFIX + "student1");
         plagiarismCase.setVerdict(PlagiarismVerdict.NO_PLAGIARISM);
         singleUserNotificationService.notifyUserAboutPlagiarismCaseVerdict(plagiarismCase, user);
         verifyRepositoryCallWithCorrectNotification(PLAGIARISM_CASE_VERDICT_STUDENT_TITLE);
         ArgumentCaptor<MimeMessage> mimeMessageCaptor = ArgumentCaptor.forClass(MimeMessage.class);
         verify(javaMailSender, timeout(1000)).send(mimeMessageCaptor.capture());
-        assertThat(mimeMessageCaptor.getValue().getSubject()).isEqualTo(PLAGIARISM_CASE_VERDICT_STUDENT_TITLE);
+        assertThat(mimeMessageCaptor.getValue().getSubject()).isEqualTo("Verdict for your plagiarism case");
+        assertThat(mimeMessageCaptor.getValue().getContent()).asString().contains("Verdict reached in plagiarism case for exercise");
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java:399]: We could use constants here for consistency

**Review by DominikRemo — APPROVED** [review]
Approve code. Just one optional nitpick

File: src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java (hunk 1 above)
Comment (jakubriegel) [reply src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java:399]: Using constant here was the reason for the bug - let's keep it simple this time ;)

**Review by dearjasmina — APPROVED** [review]
code lgtm

**Pull request merged**